components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/MetricHandler.java

```diff
@@ -0,0 +1,480 @@
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.AbstractExtendedSynapseHandler;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.API;
+import org.apache.synapse.rest.RESTUtils;
+import org.apache.synapse.transport.nhttp.NhttpConstants;
+import org.wso2.config.mapper.ConfigParser;
+import org.wso2.micro.integrator.core.internal.MicroIntegratorBaseConstants;
+import org.wso2.micro.integrator.obsrvability.handler.util.MetricConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for instrumenting Prometheus Metrics.
+ */
+public class MetricHandler extends AbstractExtendedSynapseHandler {
+    private static final String METRIC_REPORTER = "metric_reporter";
+    private static final String DELIMITER = "/";
+    private static final String EMPTY = "";
+
+    private RESTUtils restUtils = new RESTUtils();
+    private MetricReporter metricReporterInstance;
+    private Class loadedMetricClass;
+    private static Log log = LogFactory.getLog(MetricHandler.class);
+
+    @Override
+    public boolean handleInit() {
+        getMetricReporter();
+        metricReporterInstance.initMetrics();
+
+        String host = System.getProperty(MicroIntegratorBaseConstants.LOCAL_IP_ADDRESS);
+        String port = System.getProperty(MetricConstants.HTTP_PORT);
+        String javaVersion = System.getProperty(MetricConstants.JAVA_VERSION);
+        String javaHome = System.getProperty(MetricConstants.JAVA_HOME);
+
+        metricReporterInstance.serverUp(host, port, javaVersion, javaHome);
+        return true;
+    }
+
+    /**
+     * Load the MetricReporter class from the deployment.toml file if a user has defined a MetricReporter.
+     * Use default PrometheusReporter if the user hasn't defined a MetricReporter or an error occurs
+     * during custom MetricReporter class invocation.
+     */
+    private void getMetricReporter() {
+        Map<String, Object> configs = ConfigParser.getParsedConfigs();
+        Object metricReporterClass = configs.get(MetricConstants.PROMETHEUS_HANDLER + "." + METRIC_REPORTER);
+        log.info("Loading the class " + metricReporterClass);
+
+        if (metricReporterClass != null) {
+            try {
+                loadedMetricClass = Class.forName(metricReporterClass.toString());
+                metricReporterInstance = (MetricReporter) loadedMetricClass.newInstance();
+            } catch (Exception e) {
+                log.error("Error in loading the class" + metricReporterClass.toString() +
+                        "Hence loading the default PrometheusReporter class ");
+                loadPrometheusReporter();
+            }
+        } else {
+            loadPrometheusReporter();
+        }
+    }
+
+    /**
+     * Load the PrometheusReporter class by default.
+     */
+    private void loadPrometheusReporter() {
```

Comment:
this need to be renamed as default sthng

Reply:
Fixed